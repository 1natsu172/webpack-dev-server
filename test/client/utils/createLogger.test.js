'use strict';

const createLogger = require('../../../client-src/default/utils/createLogger');

describe('log', () => {
  let consoleMock;

  beforeEach(() => {
    consoleMock = jest.spyOn(console, 'info');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should output log when use the default level', () => {
    const logger = createLogger();

    logger.info('foo');

    expect(consoleMock).toBeCalled();
    expect(consoleMock.mock.calls[0][0]).toMatchSnapshot();
  });

  it('should not output log when specify the level', () => {
    const logger = createLogger('none');

    logger.info('foo');
    expect(consoleMock).not.toBeCalled();
  });
});
