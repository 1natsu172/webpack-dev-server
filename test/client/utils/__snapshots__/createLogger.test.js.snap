// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log should output log when use the default level 1`] = `"[webpack-dev-server] foo"`;
